src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -0,0 +1,132 @@
+package org.plp.implementations.plpisa;
+
+import org.plp.isa.AsmFile;
+import org.plp.isa.AsmProgram;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an in memory representation of an asm file {@link AsmFile}.
+ * This will map to a file path and will be part of {@link PlpProgram}
+ */
+public class PlpFile implements AsmFile {
+    private final String fileName;
+    private List<String> fileContent;
+    private final AsmProgram program;
+
+    public PlpFile(String fileName, AsmProgram program) throws IOException {
+        validateFileName(fileName, program);
+        this.fileName = fileName;
+        this.program = program;
+        this.fileContent = new ArrayList<>();
+    }
+
+    /**
+     * Verify if {@link AsmProgram } already has an {@link AsmFile} with the given file name
+     * @param fileName file name which needs to be verified
+     * @param program AsmProgram in which file needs to be verified
+     * @throws IOException if program already has an {@link AsmFile} with the same name
+     */
+    private void validateFileName(String fileName, AsmProgram program) throws IOException {
```

Comment:
put this method in AsmProgram?
```suggestion
    private void validateFileName(String fileName) throws IOException {
        if (this.filesInProgram.containsKey(fileName)) {
            throw new IOException();
        }
    }
```

Reply:
Yes this will be added in the future changes to AsmProgram.